compiler/rustc_ast_lowering/src/lib.rs

```diff
@@ -556,18 +541,33 @@ impl<'a, 'hir> LoweringContext<'a, 'hir> {
     fn lower_node_id(&mut self, ast_node_id: NodeId) -> hir::HirId {
         assert_ne!(ast_node_id, DUMMY_NODE_ID);
 
-        let owner = self.current_hir_id_owner;
-        let local_id = match self.node_id_to_local_id.entry(ast_node_id) {
-            Entry::Occupied(o) => *o.get(),
+        match self.node_id_to_local_id.entry(ast_node_id) {
+            Entry::Occupied(o) => {
+                hir::HirId { owner: self.current_hir_id_owner, local_id: *o.get() }
+            }
             Entry::Vacant(v) => {
                 // Generate a new `HirId`.
+                let owner = self.current_hir_id_owner;
                 let local_id = self.item_local_id_counter;
-                self.item_local_id_counter.increment_by(1);
+                let hir_id = hir::HirId { owner, local_id };
+
                 v.insert(local_id);
-                local_id
+                self.item_local_id_counter.increment_by(1);
+
+                if local_id != hir::ItemLocalId::new(0) {
```

Comment:
`ItemLocalId(0)` is manually inserted in `with_hir_owner`. This condition should always be true and could be replaced by an assertion.

Reply:
Fixed!